lib/MySQL_Monitor.cpp

```diff
@@ -121,141 +121,65 @@ static void close_mysql(MYSQL *my) {
 }
 
 
-
-struct cmp_str {
-	bool operator()(char const *a, char const *b) const
-	{
-		return strcmp(a, b) < 0;
-	}
-};
-
 class MySQL_Monitor_Connection_Pool {
-	private:
-	pthread_mutex_t mutex;
-	int size;
-	std::map<char *, PtrArray *, cmp_str> my_connections;
-	public:
-	MySQL_Monitor_Connection_Pool();
-	~MySQL_Monitor_Connection_Pool();
+private:
+	std::mutex mutex;
+	std::map<std::pair<std::string, int>, std::vector<MYSQL*> > my_connections;
+public:
+	MySQL_Monitor_Connection_Pool() {};
+	~MySQL_Monitor_Connection_Pool() { my_connections.clear(); }
```

Comment:
Both explicit ctor and dtor could be omitted. There is no need in explicit `.clear()` here too.